src/components/login-redirect/tests/LoginRedirect.test.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
+import '@testing-library/jest-dom/extend-expect';
+
+import LoginRedirect from '../LoginRedirect';
+
+import { renderWithRouter } from '../../../utils/tests';
+
+jest.mock('@edx/frontend-platform/auth');
+
+const TEST_ENTERPRISE_SLUG = 'test-enterprise-slug';
+
+const LoginRedirectWithChild = () => (
+  <LoginRedirect>
+    <span data-testid="did-i-render" />
+  </LoginRedirect>
+);
+describe('LoginRedirect', () => {
+  beforeEach(() => {
+    getAuthenticatedUser.mockReset();
+  });
+
+  test('renders children if the user is authenticated', async () => {
+    getAuthenticatedUser.mockReturnValue({ user: { email: 'test-user' } });
+    const Component = <LoginRedirectWithChild />;
+    const { history } = renderWithRouter(Component, {
+      route: `/${TEST_ENTERPRISE_SLUG}`,
+    });
+
+    // assert children are rendered
+    expect(screen.queryByTestId('did-i-render')).toBeInTheDocument();
+
+    // assert we did NOT get redirected
+    expect(history.location.pathname).toEqual(`/${TEST_ENTERPRISE_SLUG}`);
+  });
+
+  test('does not render children (due to redirect) if the user is not authenticated', async () => {
+    const Component = <LoginRedirectWithChild />;
+    renderWithRouter(Component, {
+      route: `/${TEST_ENTERPRISE_SLUG}`,
+    });
+
+    // assert children are not rendered
+    expect(screen.queryByTestId('did-i-render')).not.toBeInTheDocument();
+    // We don't test the redirect functionality explicitly as navigation is not currently supported by jsdom
```

Comment:
`renderWithRouter` lets you make assertions for redirects; there's some prior art in `UserSubsidy.test.jsx`. Essentially, `renderWithRouter` returns you a `history` object that you can assert on.

```
const { history } = renderWithRouter(Component, {
  route: `/${TEST_ENTERPRISE_SLUG}`,
});
```

Reply:
I think that only works for redirects done through the react router `Redirect` component. Redirects done through `global.location.href = x` or `global.location.assign(x)` don't seem to show up in history and give the error:
```
console.error
      Error: Not implemented: navigation (except hash changes)
```

Reply:
Wonder if [this approach](https://stackoverflow.com/a/44563899/9578634) could help?

```
<Route path={global.location} component={() => { 
     window.location.assign(proxyLoginUrl); 
     return null;
}}/>
```

Might be worth a shot. If it doesn't help, I wonder if we could mock `global.location.assign` in the test to assert that it gets called with the expected `proxyLoginUrl`?

Reply:
First approach didn't work unfortunately (same thing). I was trying to mock `global.location.assign`, but the things I tried there `global.location.assign = jest.fn()`, `jest.spyOn(global.location, 'assign')`, `jest.mock(global, ()=> { location: { assign: jest.fn() } })` and more all didn't work either for various things (usually cannot assign to read only property `assign`)
I might be missing something though